c7n/resources/s3.py

```diff
@@ -634,7 +634,9 @@ class AttachLambdaEncrypt(BucketActionBase):
                   - attach-encrypt
     """
     schema = type_schema(
-        'attach-encrypt', role={'type': 'string'})
+        'attach-encrypt',
+        role={'type': 'string'},
+        via={'type': 'string', 'enum': ['bucket-notification', 'sns']})
```

Comment:
Okay. How about `cloud-function` and `topic` per the XML examples in [this doc](http://docs.aws.amazon.com/AmazonS3/latest/dev/NotificationHowTo.html)?